### DISCUSSION TITLE
Error generated from tsml_pdf action

### DISCUSSION THREAD

**Discussion by jcookson-infoblox** [code4recovery/12-step-meeting-list#502] [General]
Writing this up in case others have seen this error ...

**Describe the bug**
Getting a critical error from tsml_pdf action,
url = https://area72aa.org/wp-admin/admin-ajax.php?action=tsml_pdf&width=4&height=7

**To Reproduce**
1. Go to 'area72aa.org'
2. Login & then select 'Dashboard' to load WP Admin
3. On left side nav, select 'Meetings'
4. Then select 'Import & Settings' sub-selection
5. Scroll over to 'Where's My Info' section
6. Click '4x7' link

**Generated Error**
```
Notice: Undefined index: in /home/customer/www/area72aa.org/public_html/wp-content/plugins/12-step-meeting-list/includes/pdf.php on line 140

Notice: Undefined index: in /home/customer/www/area72aa.org/public_html/wp-content/plugins/12-step-meeting-list/includes/pdf.php on line 140

Notice: Undefined index: in /home/customer/www/area72aa.org/public_html/wp-content/plugins/12-step-meeting-list/includes/pdf.php on line 140

Notice: Undefined index: in /home/customer/www/area72aa.org/public_html/wp-content/plugins/12-step-meeting-list/includes/pdf.php on line 140

Notice: Undefined index: name in /home/customer/www/area72aa.org/public_html/wp-content/plugins/12-step-meeting-list/includes/pdf.php on line 108

Fatal error: Uncaught Exception: TCPDF ERROR: Some data has already been output, can't send PDF file in /home/customer/www/area72aa.org/public_html/wp-content/plugins/12-step-meeting-list/includes/tcpdf/tcpdf.php:2927 Stack trace: #0 /home/customer/www/area72aa.org/public_html/wp-content/plugins/12-step-meeting-list/includes/tcpdf/tcpdf.php(7626): TCPDF->Error('Some data has a...') #1 /home/customer/www/area72aa.org/public_html/wp-content/plugins/12-step-meeting-list/includes/ajax.php(90): TCPDF->Output('meeting-schedul...', 'I') #2 /home/customer/www/area72aa.org/public_html/wp-includes/class-wp-hook.php(303): tsml_ajax_pdf('') #3 /home/customer/www/area72aa.org/public_html/wp-includes/class-wp-hook.php(327): WP_Hook->apply_filters('', Array) #4 /home/customer/www/area72aa.org/public_html/wp-includes/plugin.php(470): WP_Hook->do_action(Array) #5 /home/customer/www/area72aa.org/public_html/wp-admin/admin-ajax.php(187): do_action('wp_ajax_tsml_pd...') #6 {main} thrown in /home/customer/www/area72aa.org/public_html/wp-content/plugins/12-step-meeting-list/includes/tcpdf/tcpdf.php on line 2927
```

**Screenshots**
Just a plain standard error output page generated.

**Application**
- 12 Step Meeting List Version == 3.11.3
- WordPress version == 5.8
- Theme == Spacious Pro Child
- PHP version == 7.4.22
- MySQL version == 5.7.32-35-log

**Laptop**
- OS = mac OS Catalina v10.15.7
- RAM = 8GB
- Browsers:
1. Chrome Version 92.0.4515.131 (Official Build - x86_64)
2. Safari Version 14.1.2 (15611.3.10.1.5, 15611)
3. Firefox Quantum Version 59.0 (64 bit)
4. Microsoft Edge Version 92.0.902.73 (Official build - x86_64)

**Additional context**
Never clicked this link before so not sure if its been around for some time or not.  Also, prior to looking at this page link, I updated a couple of plugins -- (gravity forms Version 2.5.9 & give donation plugin Version 2.12.3).

Anyways, pease advise your thoughts.

**Comment by tech2serve** [comment]
Hi @jcookson-infoblox,

Thank you for the detailed write up.

I am running Wordpress 5.8 with PHP 7.4 on my dev site, and am not seeing this error. I also installed the Spacious theme (not Pro, nor modified with a child).

Can you try disabling plugins and see if the problem clears up?
